src/treeview_sandbox.ml

```diff
@@ -79,6 +79,134 @@ module Command = struct
     Extension_commands.register
       ~id:Extension_consts.Commands.open_sandbox_documentation handler
 
+  let _generate_documentation =
+    let handler (instance : Extension_instance.t) ~args =
+      let () =
+        let open Promise.Syntax in
+        let sandbox = Extension_instance.sandbox instance in
+        let (_ : unit Promise.t) =
+          match sandbox with
+          | Sandbox.Esy (_, _)
+          | Sandbox.Global
+          | Sandbox.Custom _ ->
+            let message =
+              "\"OCaml: Generate Documentation\" command only works with OPAM \
+               sandboxes."
+            in
+            let (_ : 'a option Promise.t) =
+              Window.showErrorMessage ~message ()
+            in
+            Promise.resolve ()
+          | Sandbox.Opam (opam, switch) -> (
+            let* odig = Odig.of_opam (opam, switch) in
+            match odig with
+            | Error e ->
+              let message =
+                match e with
+                | Odig.Odig_not_installed ->
+                  "\"OCaml: Generate Documentation\": the package \"odig\" \
+                   must be installed to generate documentation."
+              in
```

Comment:
Should we propose the user to install the package ?

Reply:
Yes, I believe in another PR with proper infrastructure for package installation and proposing to install certain packages. I'd put a to-do to document that we should come back with package installation proposal.

Reply:
Isn't that already doable with [Opam.install](https://github.com/ocamllabs/vscode-ocaml-platform/blob/master/src/opam.mli#L49) ?

Reply:
Sorry for a late reply. The code for opam installing is there, but I was thinking of a uniform and reusable piece of code that could be used to propose installing various dependencies that we rely on such as ocaml-lsp-server. It would have good error reporting and propose a nice way to propose installing a particular package depending on the current sandbox (opam or esy). Does this make sense? We can add opam install dialog here as well, I was just aimingfor a good piece of code reusable in various places

Reply:
Ok I see thanks. I added a comment with a link to your reply 👆 https://github.com/ocamllabs/vscode-ocaml-platform/pull/771/commits/3a089980be0f485258b13b2604c7730e79dd02d3